Add stereotype visibility toggle button

farisd16 opened the pull request:
### Checklist
- ~~[ ] I documented the TypeScript code using JSDoc style.~~
- [x] I added multiple screenshots/screencasts of my UI changes
- ~~[ ] I translated all the newly inserted strings into German and English~~

### Motivation and Context
Closes #361 

### Description
I have added a button to the popups of Components and Subsystems in a Component diagram, and to the popups of Components and Nodes in a Deployment diagram, which toggles the visibility of the stereotype.

### Steps for Testing
1. Create a component/deployment diagram
2. Add a new component/subsystem/node
3. Open the popup of the newly added element by clicking on it and then on the pencil icon 
4. Click on the stereotype visibility toggle button
5. Assert that the button works as expected

### Test Coverage
| File | Branch | Line |
| --- | --- | --- |
| stereotype-off.tsx | 100% | 43.75% |
| stereotype-on.tsx | 100% | 100% |
| stereotype-toggle.tsx | 50% | 100% |
| main/packages/common/uml-component/uml-component-component.tsx | 75% | 100% |
| uml-component-update.tsx | 100% | 47.19% |
| main/packages/common/uml-component/uml-component.ts | 100% | 100% |
| popups.ts | 100% | 100% |
| uml-component-subsystem-component.tsx | 25% | 100% |
| uml-component-subsystem-update.tsx | 100% | 48.35% |
| uml-component-subsystem.ts | 100% | 64.58% |
| main/packages/uml-component-diagram/uml-component/uml-component-component.ts | 100% | 60.52% |
| main/packages/uml-deployment-diagram/uml-deployment-component/uml-component.ts | 100% | 62.5% |
| uml-deployment-node-component.tsx | 80% | 100% |
| uml-deployment-node-update.tsx | 100% | 97.8% |
| uml-deployment-node.ts | 100% | 66% |
| typings.ts | 100% | 100% |

### Screenshots
![stereotype-toggle](https://github.com/ls1intum/Apollon/assets/63470556/5bf7ce91-af13-4103-bbac-0e001db5de44)

FelixTJDietrich reviewed (commented):
Can you explain to my why the changes so many changes are necessary? Maybe you can remove some changes that you did 🤔 
I thought you just added the new svg icons, added a button, hid the stereotype in the elements, and persisted it. Why are the other changes necessary?

farisd16 commented on src/main/packages/common/uml-component/uml-component-update.tsx:
This is a file for the popup of a Component from the Component diagram. Prior to this file, the Component was using a defualt popup (you can see this in popups.ts)

farisd16 commented on src/main/packages/uml-component-diagram/uml-component-subsystem/uml-component-subsystem-update.tsx:
I had to create it because there was no file for a popup of the Subsystem.

farisd16 commented:
I will try to clarify my changes
- components/controls/icon: I added icons for the toggle button in this directory
- stereotype-toggle.tsx: File for the new toggle button
- packages/common/uml-component: I added a popup for the Component and made sure stereotype visibility is adjusted by the toggle (both in Deployment and Component diagram, as this is in the common directory) 
- popups.ts: Registered new popups accordingly
- packages/uml-component-diagram: Analogue to changes in packages/common/uml-component
- packages/uml-deployment-diagram: Analogue to changes in packages/common/uml-component
- typints.ts: Adjusted types accordingly
- tests: Updated snapshots

GODrums commented on src/main/packages/common/uml-component/uml-component-component.tsx:
```diff
@@ -28,12 +28,12 @@ export const UMLComponentComponent: FunctionComponent<Props> = ({ element, child
         </tspan>
       )}
-      <tspan x="50%" dy={element.stereotype ? 18 : 10} textAnchor="middle">
+      <tspan x="50%" dy={element.stereotype && element.displayStereotype ? 18 : 10} textAnchor="middle">
```
Whenever the stereotype is invisible, the vertical name alignment is off-center due to the `dy=10` offset. Same goes for the other elements using a stereotype.
Using a 0-offset looks much better here, see this comparison:
![Screenshot 2024-08-01 231650](https://github.com/user-attachments/assets/ea0735ea-4004-4526-ba46-a0bbc8473847)
![Screenshot 2024-08-01 231708](https://github.com/user-attachments/assets/17651041-e782-4d10-ab42-11d0566048be)

GODrums reviewed (commented):
Looks generally good to me. 
I added an improvment suggestion regarding the alignment whenever the stereotype is hidden.

Also have a look at the results of the Codacy Static Code Analysis, you are explicitly declaring a few types that can be inferred from the context.

FelixTJDietrich reviewed (approved):
Code and functionality look good to me now 👍

GODrums reviewed (approved):
LGTM!
